docs/proposals/scheduling/20221227-cpu-static-reservation.md

```diff
@@ -0,0 +1,176 @@
+in order to ensure the isolation of the containers and the original processes. 
+At the cpu level, we usually want normal processes and containers to run on different cpu cores to reduce/avoid the impact of their resource usage.   
+eg：   
+totoal cpu on node: 24;   
+expected performance maybe like this:   
+normal process outside k8s use: 0-11;    
+container use: 12-23   
+so for k8s, the 0-11 core is reserved and it cannot be used.
+
+#### Story 2
+As a k8s cluster administrator, set the total amount of resources to be reserved directly via `node.annotation`, just like: `1C2Gi`.
+
+#### Story 3
+As a k8s cluster administrator, set the total number of cpu cores to be reserved via kubelet, and then set the specific cpu cores to be reserved via `node.annotation`, eg: kubelet sets "4C", node.annaion sets the cores with id 0-3 not to be used.
+
+### Implementation Details
+By adding annotation to the node, we specify the resources to be reserved by the koordinator, such as memory, cpu, etc. For cpu, we can specify the total amount of cpu to be reserved, or we can specify which cores to explicitly reserve. 
+```golang
+// resources that koordinator reserved, and you can reserve other resource if needed.
+type KoordReserved struct {
+	// total cpus need to be reserved, just like "2C"
+	CPUNums      string `json:"CPUNums"`
+	// specific cpus need to be reserved, just like "0-6", it means that 0-6 core cannot be used by k8s.
+	specificCPUs string `json:"specificCPUs"`
+	// eg: "1Gi"
+	memory       string `json:"memory"`
+}
+```
+#### koordlet   
+1. When koordlet reports `NodeResourceTopology`, it updates the explicitly reserved cores in node.annntation to `NodeResourceTopology.annotation["node.koordinator.sh/reserved-cpus"]`, so that the `runtimehook` can ignore these cores when fetching the shared-pool from nodetopo.
+
+2. cpu supress:  
+it will evict some pod whose qos is not lse. we should ignore the cpus reserved by node when supress with cpuset policy.  
+
+```
+cpuset(lsr) = processor(lsr) - cpusReservedByNode
+cpuset(!lse/lser) = processor(!lse/lser) - cpusReservedByNode
+``` 
+
+3. runtimehooks:  
+it will ignored the cpus reserved by `node.annotations` when create container. 
+```
+cpus(allocatable) = cpus(shared-pools) - cpus(reservedByNode)
+```  
+  
+<font color=Chocolate>*attention:*</font> The cpu occupied by the lse pod on the node is not considered here for now, but it will be removed later in the cpuset suppression phase.
+
+
+#### koord-manager  
+should update batch-resource on node, just like `batch-memory`,`batch-cpu`.the implementation of this part depends on the `nodeResource`  controller in koord-manager, so for batch-resource, the `batchResourceFit` plugin can take into account the resources reserved by the node.
+
+```
+Node(BE).Alloc = Node.Total - Node.Reserved - node.anno.reserved - System.Used - Pod(LS).Used
```

Comment:
Yes, we have to subtract the `node.Anno.Reserved` for `BE.Alloc`. We should also avoid the “double calculation” in `System.Used` since the `System.Used` is the difference between `Node.Used` with `Pod.Used`, which includes the usage of reserved resources.

Reply:
@lucming This issue remains. We may solve the problem by differentiating the usage of the node resource reserved and the system usage excluding the node resource reserved (however... it requires the koordlet to collect and report the usage of the node resource reserved). Then we can calculate the BE allocatable as below:

```
Node(BE).Alloc = Node.Total - Node.Reserved - Pod(LS).Alloc - Node.Anno.Reserved.Alloc - (System.Used - Node.Anno.Reserved.Used)
```